src/WebJobs.Script/Host/FunctionMetadataProvider.cs

```diff
@@ -193,86 +153,10 @@ private static FunctionMetadata ParseFunctionMetadata(string functionName, JObje
                     throw new FormatException($"Illegal value '{isDirectValue}' for 'configurationSource' property in {functionMetadata.Name}'.");
                 }
             }
-
+            functionMetadata.ScriptFile = Utility.DeterminePrimaryScriptFile((string)configMetadata["scriptFile"], scriptDirectory, fileSystem);
```

Comment:
@fabiocav / @soninaren - Kept parsing files and building function metadata at the application level. Validation Error if script file is not set is done in FunctionMetadataManager. Please take a look at the last two commits and let me know if you have any concerns